### PR TITLE
Rntbd added centralized monitoring for RntbdChannelEndpoint

### PR DESCRIPTION
**Pull request by moderakh**
PR moves the monitoring to RntbdClientChannelPoolMonitoringProvider

sample log

2020-09-16 15:50:19,703       [cosmos-rntbd-monitoring-rntbd-endpoints[1]] WARN  com.azure.cosmos.implementation.directconnectivity.rntbd.RntbdServiceEndpoint$RntbdEndpointMonitoringProvider - RntbdEndpoint Identifier [RntbdEndpoint, id 1, remoteAddress cdb-ms-prod-southcentralus1-fd8.documents.azure.com:14048, creationTime 2020-09-16T22:50:12.533437Z, hashCode 1508248836], Stat [ poolTaskExecutorSize 0, lastRequestNanoTime 2020-09-16T22:50:19.694037151Z, connecting 0, acquiredChannel 0, availableChannel 2, pendingAcquisitionSize 9982, closed false ]

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdServiceEndpoint.java [hunk 1]
```diff
@@ -437,5 +479,90 @@ private void evict(final RntbdEndpoint endpoint) {
...
+        }
+
+        private void logEndpoint(RntbdEndpoint endpoint) {
+            int executorPoolSize = endpoint.executorTaskQueueMetrics();
+
+            if (this.logger.isDebugEnabled()) {
+                logger.debug("RntbdEndpoint Identifier {}, Stat {}", getPoolId(endpoint), getPoolStat(endpoint));
+            }
+
+            if (executorPoolSize > 5_000 ||
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdServiceEndpoint.java:530]: else if

**Review by FabianMeiswinkel — APPROVED** [review]

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdServiceEndpoint.java (hunk 1 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdServiceEndpoint.java]: ah right. Thanks will fix.

Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdServiceEndpoint.java]: thanks addressed.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java [hunk 2]
```diff
@@ -274,6 +264,38 @@ public int channelsAvailableMetrics() {
         return this.availableChannels.size();
     }
 
+    /**
+     * Gets the number of connections which are getting established.
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java:268]: Incorrect javadoc

**Review by kushagraThapar — APPROVED** [review]
LGTM

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java (hunk 2 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java:268]: thanks. addressed.

**Comment by mitchdenny** [comment]
Merged at @kushagraThapar's request.

**Pull request merged**